cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/WorkflowStatsSLAHttpHandler.java

```diff
@@ -89,4 +111,181 @@ public void workflowStats(HttpRequest request, HttpResponder responder,
     }
     responder.sendJson(HttpResponseStatus.OK, workflowStatistics);
   }
+
+  @GET
+  @Path("apps/{app-id}/workflows/{workflow-id}/runs/{run-id}/statistics")
+  public void workflowRunDetail(HttpRequest request, HttpResponder responder,
+                                @PathParam("namespace-id") String namespaceId,
+                                @PathParam("app-id") String appId,
+                                @PathParam("workflow-id") String workflowId,
+                                @PathParam("run-id") String runId,
+                                @QueryParam("limit") int limit,
+                                @QueryParam("interval") String interval) throws Exception {
+    if (limit < 0) {
+      throw new BadRequestException("Count has to be greater than or equal to 0");
+    }
+
+    long timeInterval = TimeMathParser.resolutionInSeconds(interval);
+    Id.Workflow workflow = Id.Workflow.from(Id.Namespace.from(namespaceId), appId, workflowId);
+    Collection<WorkflowDataset.WorkflowRunRecord> workflowRunRecords =
+      store.retrieveSpacedRecords(workflow, runId, limit, timeInterval).values();
+
+    List<WorkflowRunMetrics> workflowRunMetricsList = new ArrayList<>();
+    Map<String, Long> startTimes = new HashMap<>();
+    for (WorkflowDataset.WorkflowRunRecord workflowRunRecord : workflowRunRecords) {
+      workflowRunMetricsList.add(getDetailedRecord(workflow, workflowRunRecord.getWorkflowRunId()));
+      startTimes.put(workflowRunRecord.getWorkflowRunId(),
+                     RunIds.getTime(RunIds.fromString(workflowRunRecord.getWorkflowRunId()), TimeUnit.SECONDS));
+    }
+
+    Collection<WorkflowStatsComparison.ProgramNodes> formattedStatisticsMap = format(workflowRunMetricsList);
+    responder.sendJson(HttpResponseStatus.OK, new WorkflowStatsComparison(startTimes, formattedStatisticsMap));
+  }
+
+  @GET
+  @Path("apps/{app-id}/workflows/{workflow-id}/run-id/{run-id}/compare")
+  public void compare(HttpRequest request, HttpResponder responder,
+                      @PathParam("namespace-id") String namespaceId,
+                      @PathParam("app-id") String appId,
+                      @PathParam("workflow-id") String workflowId,
+                      @PathParam("run-id") String runId,
+                      @QueryParam("other-run-id") String otherRunId) throws Exception {
+    Id.Workflow workflow = Id.Workflow.from(Id.Namespace.from(namespaceId), appId, workflowId);
+    WorkflowRunMetrics detailedStatistics = getDetailedRecord(workflow, runId);
+    WorkflowRunMetrics otherDetailedStatistics = getDetailedRecord(workflow, otherRunId);
+    if (detailedStatistics == null) {
+      responder.sendString(HttpResponseStatus.NOT_FOUND, "The run-id provided was not found : " + runId);
```

Comment:
nit: since you are using exceptions everywhere in this class, throw a ``NotFoundException`` here too?